pkg/openshift/certgen/config_test.go

```diff
@@ -0,0 +1,134 @@
+package certgen
+
+import (
+	"fmt"
+	"net"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+	"testing"
+
+	"github.com/Azure/acs-engine/pkg/openshift/filesystem"
+)
+
+var fsdata = make(map[string]datastore)
+
+type datastore struct {
+	user  string
+	group string
+	mode  os.FileMode
+}
+
+type fakefilesystem struct{}
+
+func (fakefilesystem) WriteFile(filename string, data []byte, fi filesystem.Fileinfo) error {
+	// check that you've been asked to write the given file with sane permissions & owner
+	// store the incoming files so we can compare them later on
+	fsdata[filename] = datastore{fi.User, fi.Group, fi.Mode}
+	// fmt.Printf("Filename: %v	User: %s	Group: %s	Permissions: %04o   string: %v\n", filename, fi.User, fi.Group, fi.Mode, fi.Mode.String())
+	return nil
+}
+
+func (fakefilesystem) Mkdir(filename string, fi filesystem.Fileinfo) error {
+	fmt.Printf("Filename: %v	User: %s	Group: %s	Permissions: %04o   string: %v\n", filename, fi.User, fi.Group, fi.Mode, fi.Mode.String())
+
+	fsdata[filename] = datastore{fi.User, fi.Group, fi.Mode}
+	return nil
+}
+
+func (fakefilesystem) Close() error {
+	return nil
+}
+
+var _ filesystem.Filesystem = &fakefilesystem{}
+
+func TestConfigFilePermissions(t *testing.T) {
+	c := Config{
+		templates: templates{},
+		Master: &Master{
+			Hostname: fmt.Sprintf("%s-master-%s-0", "test", "test"),
+			IPs: []net.IP{
+				net.ParseIP("10.0.0.1"),
+			},
+		},
+	}
+
+	err := c.PrepareMasterCerts()
+	if err != nil {
+		t.Fatal(err)
+	}
+	err = c.PrepareMasterKubeConfigs()
+	if err != nil {
+		t.Fatal(err)
+	}
+	err = c.PrepareMasterFiles()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	err = c.PrepareBootstrapKubeConfig()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// create mock filesystem
+	fs := fakefilesystem{}
+
+	err = c.WriteMasterKubeConfigs(fs)
```

Comment:
can you replace all of this with `err = c.WriteMaster(fs)` ?

Reply:
This worked and was much cleaner.